pkg/asset/installconfig/aws/validation_test.go

```diff
@@ -168,6 +185,53 @@ func TestValidate(t *testing.T) {
 		}(),
 		availZones:     validAvailZones(),
 		privateSubnets: validPrivateSubnets(),
+	}, {
+		name: "valid instance types",
+		installConfig: func() *types.InstallConfig {
+			c := validInstallConfig()
+			c.Platform.AWS = &aws.Platform{Region: "us-east-1"}
+			c.Platform.AWS.DefaultMachinePlatform.InstanceType = "m5.xlarge"
```

Comment:
@abhinavdahiya  This is causing a panic in unit tests and I can't figure out what I'm doing wrong.

Reply:
Pushed an update that works, but I want to understand why.

Reply:
I think i figured it out.  For compute and controlplane, &aws.MachinePool was already instantiated in valdiInstallConfig.  It is not the case for DefaultmachinePlatform, so direct assignment of InstanceType was failing.